### PR TITLE
Salt Cloud EC2: Image Name

### PR DESCRIPTION
**Pull request by pakdel**
### What does this PR do?
Add the ability to use Amazon EC2 AMI name, instead of Image ID

### Previous Behavior
`image` had to be AMI ID in form of `ami-xxxxxxxx`

### New Behavior
`image` can be and AMI name, like `CentOS Linux 7 x86_64 HVM EBS 1801_01`, or a glod like `CentOS Linux 7 *`

### Tests written?

No

### Commits signed with GPG?

No

### CODE REVIEW

File: salt/cloud/clouds/ec2.py [hunk 1]
```diff
@@ -1191,6 +1191,33 @@ def get_tenancy(vm_):
...
 
 
+def get_imageid(vm_):
+    '''
+    Returns the ImageId to use
+    '''
+    image = config.get_cloud_config_value(
+        'image', vm_, __opts__, search_global=False
+    )
+    if image.startswith('ami-'):
```
Comment (pakdel) [inline salt/cloud/clouds/ec2.py]: Should we use a regex instead?

File: doc/topics/cloud/aws.rst [hunk 2]
```diff
@@ -721,6 +721,24 @@ them have never been used, much less tested, by the Salt Stack team.
 .. __: https://aws.amazon.com/marketplace
 
 
+NOTE: If ``image`` of a profile does not start with ``ami-``, latest
+image with that name will be used.
```
Comment (cachedout) [inline doc/topics/cloud/aws.rst]: Could we add an example of this in the documentation just so it's extremely clear?

File: salt/cloud/clouds/ec2.py (hunk 1 above)
Comment (cachedout) [reply salt/cloud/clouds/ec2.py]: I'm not sure why we would need to.

File: doc/topics/cloud/aws.rst (hunk 2 above)
Comment (cachedout) [reply doc/topics/cloud/aws.rst]: If we can get these requested docs added, then I think we can get this merged.

**Review by cachedout — APPROVED** [review]

**Comment by pakdel** [comment]
Thank you @cachedout

**Pull request merged**